opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/authorization/ServiceSessionAndAuthSessionTest.java

```diff
@@ -0,0 +1,244 @@
+    private static final UUID REQUEST_ID = UUID.fromString("e3865c6b-70f2-4c1e-ad31-d7c2ff160858");
+    private static final String REDIRECT_URL_OK = "http://google.com";
+    private static final String REDIRECT_URL_NO_OK = "http://microsoft.com";
+
+    @Autowired
+    private ListAccountsService listAccountsService;
+
+    @Autowired
+    private UpdateAuthorizationService updateAuthorizationService;
+
+    @Autowired
+    private ServiceSessionRepository serviceSessionRepository;
+
+    @Autowired
+    private AuthenticationSessionRepository authenticationSessions;
+
+    @MockBean
+    private Xs2aListAccountsEntrypoint xs2aListAccountsEntrypoint;
+
+    @MockBean
+    private Xs2aUpdateAuthorization xs2aUpdateAuthorization;
+
+    @Test
+    @SneakyThrows
+    void serviceSessionAndAuthSessionWithConsent_success() {
+        UUID sessionId = UUID.randomUUID();
+
+        // STEP 1
+        ValidationErrorResult validationErrorResult = buildValidationErrorResultResult();
+
+        doAnswer(invocation -> CompletableFuture.completedFuture(validationErrorResult))
+                .when(xs2aListAccountsEntrypoint)
+                .execute(any());
+
+        FacadeStartAuthorizationResult listAccountsResponse = (FacadeStartAuthorizationResult) listAccountsService.execute(buildListAccountRequest(sessionId)).get();
+
+        assertThat(UUID.fromString(listAccountsResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(listAccountsResponse.getServiceSessionId())).isEqualTo(sessionId);
+        assertThat(listAccountsResponse.getRedirectionTo()).isEqualTo(validationErrorResult.getRedirectionTo());
+
+        await().atMost(Durations.ONE_SECOND)
+                .pollDelay(Durations.ONE_HUNDRED_MILLISECONDS)
+                .until(() -> authenticationSessions.findAll().iterator().hasNext());
+
+        assertAllSessions(sessionId);
+
+        // STEP 2
+        AuthorizationRequiredResult authorizationRequiredResult = buildAuthorizationRequiredResult();
+
+        doAnswer(invocation -> CompletableFuture.completedFuture(authorizationRequiredResult))
+                .when(xs2aUpdateAuthorization)
+                .execute(any());
+
+        FacadeRedirectResult authUpdatedResult = (FacadeRedirectResult) updateAuthorizationService.execute(buildAuthRequest(listAccountsResponse)).get();
+
+        assertThat(UUID.fromString(authUpdatedResult.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(authUpdatedResult.getServiceSessionId())).isEqualTo(sessionId);
+
+        assertThat(authUpdatedResult.getRedirectionTo()).isEqualTo(authorizationRequiredResult.getRedirectionTo());
+
+        assertAllSessions(sessionId);
+    }
+
+    @Test
+    @SneakyThrows
+    void serviceSession_protocolError() {
+        UUID sessionId = UUID.randomUUID();
+
+        ErrorResult<AuthorizationRequiredResult> errorResult = new ErrorResult<>();
+        errorResult.setCode("400");
```

Comment:
As it is not parsed anywhere, maybe it is better not to set it

Reply:
error code and error message are checked in assertErrorResponse() method